client/app/pods/components/type-ahead/component.coffee

```diff
@@ -0,0 +1,54 @@
+`import Ember from 'ember'`
+
+TypeAheadComponent = Ember.TextField.extend
+
+  suggestionEngine: null
+  displayKey: null
+  selection: null
+  url: null
+
+  setupSuggestionEngine:(->
+    self = @
+    @_super()
+    @suggestionEngine = new Bloodhound(
+      datumTokenizer: Bloodhound.tokenizers.obj.whitespace(self.get('displayKey'))
+      queryTokenizer: Bloodhound.tokenizers.whitespace
+      remote:
+        url: self.get('url')
+        filter: (response) ->
+          $.map response.filtered_users, (user) ->
+            { name: user.full_name, email: user.email }
+    )
+
+    @suggestionEngine.initialize()
+  ).on('init'),
+
+  setup:(->
+    @.$().typeahead {
+      hint: true
+      highlight: true
+      minLength: 2
+    },
+      name: 'auto-complete'
+      displayKey: @get('displayKey')
+      source: @get('suggestionEngine').ttAdapter()
+      templates:
+        empty: ->
+          "no results"
+        suggestion: (context) ->
+          "#{context.name} #{context.email}"
+
+
+    @.$().on('typeahead:selected', (obj, data, name) ->
+      console.log obj
+      console.log data
+      console.log name
+    )
+
+  ).on('didInsertElement'),
```

Comment:
Everything inside the `setup` method needs to be wrapped inside an `Ember.run.schedule('afterRender', ->)`, because the setup function may be called before the browser renders the children of the component DOM element that you have just inserted. There's more information here: http://mavilein.github.io/javascript/2013/08/01/Ember-JS-After-Render-Event/